starter-code/routes/auth-routes.js

```diff
@@ -0,0 +1,100 @@
+const express = require("express");
+const authRoutes = express.Router();
+const User = require("../models/user");
+const bcrypt = require("bcrypt");
+const bcryptSalt = 10;
+
+//redirect to signup page
+authRoutes.get("/signup", (req, res, next) => {
```

Comment:
this router handles `/signup` - both GET and POST - as well as `/login` - also both GET and POST

you should have a `.use(...)` middleware at the top, before the actual routes, so that you can authorize users only if they ARE NOT logged in
 
```
authRoutes.use(function (req, res, next) {
  if (!req.session.currentUser) {
    next(); // we DO NOT have a user in the session, continue executing routes
  } else {
    res.redirect('/private'); // user is logged in, redirect to /welcome
  }
});
```